src/tonic/api/points_internal_api.rs

```diff
@@ -37,9 +38,12 @@ impl PointsInternalService {
 impl PointsInternal for PointsInternalService {
     async fn upsert(
         &self,
-        request: Request<UpsertPointsInternal>,
+        mut request: Request<UpsertPointsInternal>,
     ) -> Result<Response<PointsOperationResponseInternal>, Status> {
         validate_and_log(request.get_ref());
+
+        let claims = request.extensions_mut().remove::<Claims>();
```

Comment:
I suggest to put code that calls `extensions_mut().insert()` and `extensions_mut().remove()` to the same module. Then this line should look like this:

```suggestion
        let claims = super::auth::extract_claims(&mut request);
```